rust-keybroker/keybroker-server/src/verifier.rs

```diff
@@ -68,7 +74,14 @@ pub fn verify_with_veraison_instance<DE: EmitDiagnostic>(
     diagnostics: &DE,
 ) -> Result<bool> {
     // Get the discovery URL from the base URL
-    let discovery = Discovery::from_base_url(String::from(verifier_base_url))?;
+    let discovery_api_endpoint = format!(
+        "{}{}",
+        verifier.base_url, "/.well-known/veraison/verification"
```

Comment:
The intention behind the `Discovery` interface was that it would encapsulate the knowledge of the `.well-known` sub-paths inside the API client as a convenience for consumers. If we need to supply that path explicitly here, then I think that's a usability failing in the Rust client. Was this changed when HTTPs was introduced? If so, then we might need to re-think that.

Reply:
Agreed, I took what came from the rust-apiclient example, but will check if there are other constructors method --- use a `.with_base_url` method rather than the `.with_url `one.

Reply:
[https://github.com/veraison/rust-apiclient/pull/21](https://github.com/veraison/rust-apiclient/pull/21) has been submitted to address this.

Reply:
> [veraison/rust-apiclient#21](https://github.com/veraison/rust-apiclient/pull/21) has been submitted to address this.

Thank you!  I've just merged it.

Reply:
Thank you !

I'm using it in the next iteration of this patch, but I believe the API could be tweaked a tiny bit so it becomes easier / more elegant to use --- e.g. the `base_url` is mandatory, so I would tend to have it as an argument to the `new` function. And maybe `with_root_certificate` should take an Option<> to avoid spaghetti code at the caller's site (arguably, this seems a generic pattern enough that rust could have some solutions for that)